### PR TITLE
Ignore duplicate keys

### PR DESCRIPTION
**Pull request by dalepsmith**
https://github.com/keepsimple1/mdns-sd/issues/74

### CODE REVIEW

**Comment by dalepsmith** [comment]
Actually #74

**Comment by keepsimple1** [comment]
> #74

Shouldn't it be a link to the issue?

File: src/service_info.rs [hunk 1]
```diff
@@ -302,7 +302,7 @@ fn decode_txt(txt: &[u8]) -> HashMap<String, String> {
                 Some(idx) => {
                     let k = &kv_string[..idx];
                     let v = &kv_string[idx + 1..];
-                    kv_map.insert(k.to_string(), v.to_string());
+                    kv_map.entry(k.to_string()).or_insert_with(|| v.to_string());
```
Comment (keepsimple1) [inline src/service_info.rs:305]: nit: why not use `or_insert(v.to_string())`?

**Review by keepsimple1 — COMMENTED** [review]
Looks good. Thanks! A nit inline.

**Comment by dalepsmith** [comment]
On 11/11/22, keepsimple1 ***@***.***> wrote:
> @keepsimple1 commented on this pull request.
>
> Looks good. Thanks! A nit inline.
>
>> @@ -302,7 +302,7 @@ fn decode_txt(txt: &[u8]) -> HashMap<String, String>
>> {
>                  Some(idx) => {
>                      let k = &kv_string[..idx];
>                      let v = &kv_string[idx + 1..];
> -                    kv_map.insert(k.to_string(), v.to_string());
> +                    kv_map.entry(k.to_string()).or_insert_with(||
> v.to_string());
>
> nit: why not use `or_insert(v.to_string())`?

Because Clippy didn't like it.  ;^}

WIth `or_insert`, the `to_string` is called every time, if it's needed
or not. With `or_insert_with`, `to_string` is *only* called when
actually doing the insert.

-Dale

**Comment by dalepsmith** [comment]
On 11/11/22, keepsimple1 ***@***.***> wrote:
>> #74
>
> Shouldn't it be a link to the issue?

Sorry.  Still new at doing this kind of thing though the github web interface.

-Dale

**Comment by keepsimple1** [comment]
> WIth `or_insert`, the `to_string` is called every time, if it's needed
or not. With `or_insert_with`, `to_string` is *only* called when
actually doing the insert.

Good point! Indeed, such pattern is quite common in Rust.

**Review by keepsimple1 — APPROVED** [review]

**Pull request merged**